### DISCUSSION TITLE
Completion in derive macro attributes

### DISCUSSION THREAD

**Discussion by mikatpt** [rust-lang/rust-analyzer#15555] [Ideas]
This might be a naive question/suggestion. Is it possible to have completion of derive attributes, and functional completion within their values?

For example, `clap`'s derive macro has a huge list of undocumented attributes which are drawn from its `builder` API's public interface. If it weren't for the lack of completion, it would feel like an elegant, very well-designed API—as it is, devX is extremely lackluster compared to the `builder` API because rust-analyzer doesn't help you out.

```rust
#[command(
    name = "some-cli-program",   // suggest attribute titles?
    color = ColorChoice::Always, // auto-complete doesn't function here.
    styles = Styles::styled()
        .header(AnsiColor::Yellow.on_default())
        .usage(AnsiColor::Yellow.on_default())
        .literal(AnsiColor::Magenta.on_default())
        .placeholder(AnsiColor::Green.on_default())
)]
```

**Comment by lnicola** [comment]
Related: https://github.com/rust-lang/rust-analyzer/discussions/15452.
